Objective-C/CBLDatabase.mm

```diff
@@ -624,9 +655,30 @@ - (nullable NSDate*) getDocumentExpirationWithID: (NSString*)documentID {
 #pragma mark - INTERNAL
 
 - (void) mustBeOpen {
-    if (_c4db == nullptr) {
+    if ([self isClosed])
         [NSException raise: NSInternalInconsistencyException
                     format: @"%@", kCBLErrorMessageDBClosed];
+}
+
+- (void) mustBeOpenLocked {
```

Comment:
do you think this will be used other than in CBLQuery? 

if only used in CBLQuery, can we directly lock at the caller
```
CBL_LOCK(self.database) { [self.database mustBeOpen]; } 
```

Reply:
`mustBeOpen` is called in many places inside the database classes from the lock, for consistency can we include lock inside base method `mustBeOpen` itself and remove `mustBeOpenLocked`.

Reply:
mustBeOpen is already called inside a lock so the purpose of not having lock here because it doesn't need to lock again.